src/Catalyst.Core.Modules.Consensus/Deltas/Building/CandidateBuilderStep.cs

```diff
@@ -0,0 +1,127 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Catalyst.Abstractions.Cryptography;
+using Catalyst.Abstractions.Hashing;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.Util;
+using Catalyst.Core.Modules.Dfs.Extensions;
+using Catalyst.Protocol.Peer;
+using Catalyst.Protocol.Transaction;
+using Catalyst.Protocol.Wire;
+using Google.Protobuf;
+using LibP2P;
+using Nethermind.Dirichlet.Numerics;
+using TheDotNetLeague.MultiFormats.MultiBase;
+
+namespace Catalyst.Core.Modules.Consensus.Deltas.Building
+{
+    internal sealed class CandidateBuilderStep : IDeltaBuilderStep
+    {
+        private readonly PeerId _producerUniqueId;
+        private readonly IDeterministicRandomFactory _randomFactory;
+        private readonly IHashProvider _hashProvider;
+
+        public CandidateBuilderStep(PeerId producerUniqueId,
+            IDeterministicRandomFactory randomFactory,
+            IHashProvider hashProvider)
+        {
+            _producerUniqueId = producerUniqueId ?? throw new ArgumentNullException(nameof(producerUniqueId));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name '' is not declared in the argument list.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)